### PR TITLE
docs: updated pattern template

### PR DESCRIPTION
**Pull request by connor-leech**
## What’s changed?

- It’s now called “How to write a pattern” so that it’s more action-oriented and can do double duty as guidance on the website, and also serve as content for the Markdown template.
- The sections have been finessed and sentences have been sharpened.
- We kept the amount of detailed guidance to bare bones. A challenge has been getting the balance right between detailed guidance and making people feel overwhelmed. For the fine-tuned detail esp. on images, we’ll plan to be in a good place with production guidelines in the future and let that space do the heavy lifting for us.

### CODE REVIEW

**Comment by janchild** [comment]
@connor-leech, let's do one more round. I'll send you a meeting invitation.

File: src/pages/how-to-contribute/pattern.mdx [hunk 1]
```diff
@@ -1,63 +1,102 @@
 ---
-title: Patterns
+title: How to write a pattern
```
Comment (connor-leech) [inline src/pages/how-to-contribute/pattern.mdx]:
```suggestion
title: Patterns
```

File: docs/pattern-template.mdx [hunk 2]
```diff
@@ -1,55 +1,94 @@
...
+Evaluate your pattern to ensure it meets [accessibility standards](/guidelines/accessibility/overview) and guidelines,
+and provide details of compliance.
+
+For example, “Users should be able to TAB into the input field of the search box
+to being typing and press ENTER to run the search query.”
+
+## References
 
-**Link to any related patterns.** This may not be the right pattern for a designer's problem, but other closely related patterns might be helpful.
+Include any citations for research, books, and articles you found helpful. These
```
Comment (janchild) [inline docs/pattern-template.mdx]:
```suggestion
Help designers understand your process by explaining your rationale for the way you implemented the pattern. Include any research, citations, books or articles that you found helpful.  
```

File: docs/pattern-template.mdx [hunk 3]
```diff
@@ -1,55 +1,94 @@
...
+help designers understand your decisions.
```
Comment (janchild) [inline docs/pattern-template.mdx]:
```suggestion
Include any material that will help designers understand your decisions.
```

File: docs/pattern-template.mdx [hunk 4]
```diff
@@ -1,55 +1,94 @@
...
+- Create a functional prototype if that is the best way to illustrate the pattern's behavior.
+- Provide a Sketch file with any symbols you've created in the development of this pattern.
 
-## Metadata 
+### Variations
 
-**Include a list of components used in the pattern.** Link to where where each component is documented.
+If there are variations that support different use cases, include those here
+with corresponding explanations and visuals. Be sure to explain the reasons for
+using one pattern over another.
```
Comment (janchild) [inline docs/pattern-template.mdx]:
```suggestion
using one solution over another.
```

File: docs/pattern-template.mdx [hunk 5]
```diff
@@ -1,55 +1,94 @@
...
+the IBM Design System.
 
-**Offer any development notes you may have.** Many patterns don't require specific developer attention to implement. If you have any developer guidance, include it here.
+- Use annotated wireframes or sketches to explain the visual hierarchy of components.
+- Create high-level user flows to explain the big picture.
+- Create a functional prototype if that is the best way to illustrate the pattern's behavior.
+- Provide a Sketch file with any symbols you've created in the development of this pattern.
 
-## Metadata 
+### Variations
```
Comment (janchild) [inline docs/pattern-template.mdx]:
```suggestion
### Variations
```
```suggestion
### Other use cases
```

File: docs/pattern-template.mdx [hunk 6]
```diff
@@ -1,55 +1,94 @@
...
 
-**Include a list of components used in the pattern.** Link to where where each component is documented.
+If there are variations that support different use cases, include those here
```
Comment (janchild) [inline docs/pattern-template.mdx]:
```suggestion
If there are use cases that require a different solution, include those here
```

File: docs/pattern-template.mdx [hunk 7]
```diff
@@ -1,55 +1,94 @@
...
+with corresponding explanations and visuals. Be sure to explain the reasons for
+using one pattern over another.
+
+## Related patterns and components
+
+Provide links to related patterns and components.
```
Comment (janchild) [inline docs/pattern-template.mdx]:
```suggestion
Provide links to related patterns and components.
```
```suggestion
Which components did you use when building this pattern? List them here. 
```

**Pull request merged**